### PR TITLE
[ContactStructuralMechanicsApplication] Minor refactor. More pythonic scripts

### PR DESCRIPTION
**Pull request by loumalouomega**
Basically more pythonic scripts

### CODE REVIEW

File: applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
+
 # Importing the Kratos Library
 import KratosMultiphysics as KM
 import KratosMultiphysics.ContactStructuralMechanicsApplication as CSMA
 
-def print_on_rank_zero(*args):
+# Import sys
+import sys
+
+def  print_on_rank_zero(*args):
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py]: This is no longer needed

File: applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
     # This function will be overridden in the trilinos-solvers
     KM.Logger.PrintInfo(" ".join(map(str, args)))
 
-def print_warning_on_rank_zero(*args):
+def  print_warning_on_rank_zero(*args):
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py]: Same

File: applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py [hunk 3]
```diff
@@ -55,7 +59,7 @@ def AuxiliarContactSettings():
 
     return contact_settings
 
-def AuxiliarExplicitContactSettings():
+def  AuxiliarExplicitContactSettings():
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py:54]: Why the space?

File: applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py [hunk 4]
```diff
@@ -242,7 +246,20 @@ def AuxiliarCreateLinearSolver(main_model_part, settings, contact_settings, line
     else:
         return linear_solver
 
-def AuxiliarLineSearch(computing_model_part, mechanical_scheme, linear_solver, mechanical_convergence_criterion, builder_and_solver, settings, contact_settings, processes_list, post_process):
+def  AuxiliarPureSlipCheck(model_part):
+    # Check if the properties have friction coefficient
+    aux_sum_friction_coeff = 0.0
+    for prop in model_part.GetProperties():
+           if prop.Has(KM.FRICTION_COEFFICIENT):
+               aux_sum_friction_coeff += prop.GetValue(KM.FRICTION_COEFFICIENT)
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py]: This is a perfect candidate for list comprehension 
Look it up, its very useful!

File: applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py [hunk 5]
```diff
@@ -242,7 +246,20 @@ def AuxiliarCreateLinearSolver(main_model_part, settings, contact_settings, line
...
+
+    # Check value
+    if aux_sum_friction_coeff > sys.float_info.epsilon:
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py]: Can be reduced to a one-liner
Btw I havent seen this epsilon being used in Kratos

**Review by philbucher — COMMENTED** [review]
Some minor comments

File: applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py (hunk 3 above)
Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py:54]: The kdevelop integrates PEP8 standards and s says so

File: applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py (hunk 4 above)
Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py]: yes I know, but I always forget: https://realpython.com/python-coding-interview-tips/

Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py]: I can live with this (I prefer to be more readable when no critical)

File: applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py (hunk 5 above)
Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py]: Can we keep it this way?, I think is more readable (and this is not a a crucial part of the code)

File: applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py (hunk 4 above)
Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py]: OK, you won

File: applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py (hunk 5 above)
Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py]: this epsilon is used several times (but it's me who uses it)

**Comment by loumalouomega** [comment]
Done

File: applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py (hunk 3 above)
Comment (philbucher) [reply applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py:54]: really? Two spaces?
I didn't know, do you have a link?

File: applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py [hunk 6]
```diff
@@ -235,14 +231,20 @@ def AuxiliarCreateLinearSolver(main_model_part, settings, contact_settings, line
...
             return linear_solver
     else:
         return linear_solver
 
-def AuxiliarLineSearch(computing_model_part, mechanical_scheme, linear_solver, mechanical_convergence_criterion, builder_and_solver, settings, contact_settings, processes_list, post_process):
+def  AuxiliarPureSlipCheck(model_part):
+    # Check if the properties have friction coefficient
+    aux_sum_friction_coeff = sum(prop.GetValue(KM.FRICTION_COEFFICIENT) for prop in model_part.GetProperties() if prop.Has(KM.FRICTION_COEFFICIENT))
+    # Check value
+    return False if aux_sum_friction_coeff > sys.float_info.epsilon else True
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py]:
```suggestion
    return aux_sum_friction_coeff < sys.float_info.epsilon
```

would be even shorter but up to you

**Review by philbucher — DISMISSED** [review]

**Review by philbucher — APPROVED** [review]
:+1:

**Pull request merged**